batch.py

```diff
@@ -703,15 +708,20 @@ def manage_concurrency_for_job(job: hb.job.Job):
             plink_job.depends_on(filter_job)
 
         plink_job.image(CELLREGMAP_IMAGE)
-        plink_job.call(
+        empty_plink = plink_job.call(
             get_promoter_variants,
             mt_path=output_mt_path,
             ht_path=anno_ht_path,
             gene_details=gene_dict[gene],
             window_size=window_size,
             plink_file=plink_file,
         )
-        genotype_jobs.append(plink_job)
+        # if plink files are empty, remove gene from list
+        # and remove job from future dependencies list
+        if empty_plink:
+            genes_of_interest.remove(gene)
+        else:
+            genotype_jobs.append(plink_job)
```

Comment:
AFAIK this is going to run into the same PythonResult problems - you won't know the Truth value here until the batch runs, so you can't make flow decisions based on it

Reply:
argh, I think you're right but I can't think of an easy solution :(

Reply:
I have tried, but am not sure I got the syntax right at all